gr-audio/lib/jack/jack_source.cc

```diff
@@ -126,7 +126,7 @@ jack_source::jack_source(int sampling_rate,
 
     set_output_multiple(d_jack_buffer_size);
 
-    assert(sizeof(float) == sizeof(sample_t));
+    static_assert(sizeof(float) == sizeof(sample_t), "sample_t must be float");
```

Comment:
Considering the error message, a check for ``std::is_same<float, sample_t>::value`` is more appropriate.